tools/lsp/preview.rs

```diff
@@ -68,10 +69,10 @@ thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default:
 
 pub fn set_contents(url: &VersionedUrl, content: String) {
     let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
-    let old = cache.source_code.insert(url.url().clone(), content.clone());
-    if cache.dependency.contains(&url.url()) {
-        if let Some(old) = old {
-            if content == old {
+    let old = cache.source_code.insert(url.url().clone(), (url.version().clone(), content.clone()));
+    if cache.dependency.contains(url.url()) {
+        if let Some((old_version, old)) = old {
+            if content == old && old_version == *url.version() {
```

Comment:
I think we wouldn't need to reload the preview if the version change but the document didn't.  Do i'd skip the test on the version there.